controllers/grants/index.js

```diff
@@ -150,85 +143,82 @@ async function checkSpelling(searchTerm, locale = 'en') {
+        const queryWithPage = { ...facetParams, ...{ page: req.query.page || 1, locale } };
+        const data = await queryGrantsApi(queryWithPage);
+
+        let searchSuggestions = false;
+        if (data.meta.totalResults === 0 && req.query.q) {
+            searchSuggestions = await checkSpelling(req.query.q, locale);
+        }
+
+        res.format({
+            // Initial / server-only search
+            html: async () => {
+                // Grab some case studies
+                const caseStudiesResponse = await contentApi.getCaseStudies({ locale });
+
+                // Shuffle the valid case studies and grab the first few
+                const caseStudies = sampleSize(caseStudiesResponse.filter(c => c.grantId), 3);
+
+                res.render(path.resolve(__dirname, './views/search'), {
+                    title: res.locals.copy.title,
+                    queryParams: isEmpty(facetParams) ? false : facetParams,
+                    grants: data.results,
+                    facets: data.facets,
+                    meta: data.meta,
+                    grantDataDates: grantsConfig.dateRange,
+                    caseStudies: caseStudies,
+                    grantNavLink: grantsConfig.grantNavLink,
+                    searchSuggestions: searchSuggestions,
+                    pagination: buildPagination(data.meta.pagination, queryWithPage, paginationLabels)
+                });
+            },
+
+            // AJAX search for client-side app
+            'application/json': () => {
+                /**
+                 * Repopulate existing app globals so Nunjucks
+                 * can read them outside of Express's view engine context
+                 */
+                const extraContext = {
+                    grants: data.results,
+                    pagination: buildPagination(data.meta.pagination, queryWithPage, paginationLabels)
+                };
+
+                const context = { ...res.locals, ...req.app.locals, ...extraContext };
+
+                nunjucks.render(path.resolve(__dirname, './views/ajax-results.njk'), context, (renderErr, html) => {
+                    if (renderErr) {
+                        Raven.captureException(renderErr);
+                        res.status(400).json({ error: 'ERR-TEMPLATE-ERROR' });
+                    } else {
+                        res.json({
+                            meta: data.meta,
+                            facets: data.facets,
+                            searchSuggestions: searchSuggestions,
+                            resultsHtml: html
+                        });
+                    }
+                });
+            }
+        });
     } catch (errorResponse) {
         return res.format({
             html: () => {
                 next(errorResponse.error);
             },
             'application/json': () => {
+                Raven.captureMessage(errorResponse);
```

Comment:
This nested `error.error` logic is still a bit funny so logging out the raw error to see if we can make this more robust.